### ISSUE TITLE
Object of type ObjectId is not JSON Serializable

### ISSUE THREAD

**Issue by Voronsky** [oasis-open/cti-taxii-server#24]
Greetings Everyone,

 Been playing around with the cti-taxii server and I've come into an issue. It seems that when running Medallion and using MongoDB, I have an issue in trying to retrieve objects from a collection and insert.

I have medallion use a config file that specifies that the backend to be used is my local mongoDB instance.

Attached is an image of Medallion output
![Imgur](https://i.imgur.com/it5q1je.jpg)

Since I was playing around with it in a python3 interpreter, the following is what I did
```python

collection = Collection("http:/127.0.0.1:5000/trustgroup1/collections/91a7b528-80eb-42ed-a74d-c6fbd5a26116/",user="admin",password="Password0")
mw = Malware(name="Test",description="Real bad malware",labels=["ransomware"])
collection.add_objects(json.loads(Bundle(mw).serialize())) #Crashes here and gives the image above
print(collection.get_objects()) #Same issue
```
Oddly enough, if I use the config.json file that uses the memory as backend, I have no problems with it at all, it works successfully.

Am I just doing a typo? or is this a bug?

**Comment by rpiazza** [comment]
Try:

collection.add_objects(json.dumps(Bundle(mw))

**Comment by Voronsky** [comment]
Says, the object cannot be serializable

![Imgur](https://i.imgur.com/YV4ENZX.jpg)

**Comment by emmanvg** [comment]
This may be related to oasis-open/cti-python-stix2#125.

**Comment by gtback** [comment]
The first error is related to MongoDB (ObjectID is Mongo's unique idenitifer, I think). The second error is (as you mention) because you need to call `serialize()`. I'll have to play around with Mongo a bit to see if I can reproduce this.

**Comment by Voronsky** [comment]
Thanks @gtback  ! Please let me know how it goes. I am a bit stumped as to how to proceed using MongoDB as the backend in the config; so in the mean time I am using the memory backend through the config file that is provided in one of the folders to continue playing around.

**Comment by emmanvg** [comment]
@Voronsky, have you modified the database in any way? Based on the latest changes we have made to `cti-python-stix2`, `cti-taxii-client` and `cti-taxii-server` I have not been able to reproduce the `Object of type ObjectId is not JSON serializable`. Can you pull the latest changes from these repos and test again. I am using this script: <br>

``` python

from stix2 import TAXIICollectionSink, ThreatActor, Malware, Bundle
from taxii2client import Collection

collection = Collection("http://127.0.0.1:5000/trustgroup1/collections/91a7b528-80eb-42ed-a74d-c6fbd5a26116/", user="admin", password="Password0")

#create TAXIICollectionSINK and push STIX content to it
tc_sink = TAXIICollectionSink(collection)

# create new STIX threat-actor
ta = ThreatActor(name="Teddy Bear",
                labels=["nation-state"],
                sophistication="innovator",
                resource_level="government",
                goals=[
                    "compromising environment NGOs",
                    "water-hole attacks geared towards energy sector",
                ])

mw = Malware(name="Test",description="Real bad malware",labels=["ransomware"])

tc_sink.add(ta)  # Add one object via tc_sink

collection.add_objects(Bundle(mw).serialize())  # Add one object via taxii2client

print(collection.get_objects())
```

**Comment by Voronsky** [comment]
@emmanvg  good point I forgot to mention that the DB was initialized via the generic script located in the test folder specifically for MongoDB. Furthermore, after pulling the latest commits from the 3 repos, I can say that your script test worked 100%; i did not get the original error anymore. I see it also properly inserted the object in my MongoDB side with the correct date too within the "objects" collection.

Thank you!
